### PR TITLE
feature: delayed tasks (Task.NotBefore)

### PR DESCRIPTION
**Pull request by mxmCherry**

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -27,6 +28,10 @@ CREATE INDEX IF NOT EXISTS idx_pgpq_tasks_created_at ON pgpq_tasks (created_at A
 
 CREATE INDEX IF NOT EXISTS idx_pgpq_tasks_priority_order ON pgpq_tasks (priority DESC, updated_at ASC);
 
+ALTER TABLE pgpq_tasks ADD COLUMN IF NOT EXISTS not_before TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT NOW();
```
Comment (dim) [inline schema.sql:31]: Instead of now, can we default to unix(0) time? Same with inserting new rows, pass zero time unless !IsZero()

**Review by dim — CHANGES_REQUESTED** [review]
Can you please update CI to use latest go pls

File: schema.sql (hunk 1 above)
Comment (mxmCherry) [reply schema.sql:31]: I don't like this idea too much, but OK, done.

File: pgpq.go [hunk 2]
```diff
@@ -136,7 +137,19 @@ func (td *TaskDetails) scan(rows interface{ Scan(...interface{}) error }) error
...
 		&td.Payload,
+		&td.NotBefore,
 		&td.CreatedAt,
 		&td.UpdatedAt,
 	)
 }
+
+// ----------------------------------------------------------------------------
+
+var unixZero = time.Unix(0, 0).UTC()
```
Comment (dim) [inline pgpq.go:148]: You can just use time.Time{}

**Review by dim — APPROVED** [review]

File: pgpq.go (hunk 2 above)
Comment (mxmCherry) [reply pgpq.go:148]: What? You said to default to unix(0), now it's time.Time{}

If you're OK with this PR - just merge.

Comment (mxmCherry) [reply pgpq.go:148]: And tag, just reminding

Comment (mxmCherry) [reply pgpq.go:148]: Actually, looks like I have write access to this repo - let me merge && tag it.

**Pull request merged**